mirrord/cli/src/internal_proxy.rs

```diff
@@ -47,6 +46,8 @@ fn print_port(listener: &TcpListener) -> Result<()> {
         .map_err(InternalProxyError::LocalPortError)?
         .port();
     println!("{port}\n");
+    // "leak" the `Gag` in order to redirect stdout to /dev/null for the whole run.
+    std::mem::forget(Gag::stdout());
```

Comment:
Gag still holds stdout since it's meant for temporary redirection, which means we don't close stdout which means it might make caller process still wait for it's stdout to finish.

Reply:
I would copy daemonize code that does the dup - afaik if you use dup2 it closes the original fd

> If the file descriptor newfd was previously open, it is closed
       before being reused; the close is performed silently (i.e., any
       errors during the close are not reported by dup2()).

So you actually do `dup2(dev_null_fd, stdout)`

Reply:
Also I would still do it for stdin and stderr even though the caller does `/dev/null` just to be on the safe side.